homeassistant/components/devolo_home_control/config_flow.py

```diff
@@ -36,15 +40,19 @@ async def async_step_user(self, user_input=None):
         except CredentialsInvalid:
             return self._show_form(step_id="user", errors={"base": "invalid_auth"})
 
-    async def async_step_zeroconf(self, discovery_info: DiscoveryInfoType):
+    async def async_step_zeroconf(
+        self, discovery_info: DiscoveryInfoType
+    ) -> FlowResult:
         """Handle zeroconf discovery."""
         # Check if it is a gateway
         if discovery_info.get("properties", {}).get("MT") in SUPPORTED_MODEL_TYPES:
             await self._async_handle_discovery_without_unique_id()
             return await self.async_step_zeroconf_confirm()
         return self.async_abort(reason="Not a devolo Home Control gateway.")
 
-    async def async_step_zeroconf_confirm(self, user_input=None):
+    async def async_step_zeroconf_confirm(
+        self, user_input: ConfigType | None = None
```

Comment:
ConfigType is used for YAML configuration.,

```suggestion
        self, user_input: dict[str, Any] | None = None
```